pkg/library/ssh/event.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019-2024 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ssh
+
+import (
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/devworkspace-operator/pkg/constants"
+	"github.com/devfile/devworkspace-operator/pkg/library/lifecycle"
+)
+
+// AddSshAgentPostStartEvent Start ssh-agent and add the default ssh key to it, if the ssh key has a passphrase.
+// Initialise the ssh-agent session env variables in the user .bashrc file.
+func AddSshAgentPostStartEvent(spec *v1alpha2.DevWorkspaceTemplateSpec) error {
+	if spec.Commands == nil {
+		spec.Commands = []v1alpha2.Command{}
+	}
+
+	if spec.Events == nil {
+		spec.Events = &v1alpha2.Events{}
+	}
+
+	var commandLine = `SSH_ENV_PATH=/home/user/ssh-environment \
```

Comment:
A few comments:
- The `commandLine` variable could probably be a constant to make the rest of the code a little easier to read (like we did [here](https://github.com/devfile/devworkspace-operator/pull/1251/files#diff-f1f3498fd50bbc235d4d09f5f5485a91f5a4a5cfeb20b127c96fc849a30ba56bR30-R46))
- It might be better to use `$HOME` instead of hard-coding `/home/user/` since the users workspace might not be using the UDI. I'm not sure how common the `ssh-agent` binary is, but hopefully your implementation here could also work for other workspace images.
- This is very subjective: I find using `source` is more readable than `.`. This is relevant IMO because we are modifying the users `.bashrc`.


The only non-straightforward thing here is the  `ssh-agent | sed 's/^echo/#echo/' > $SSH_ENV_PATH`. My understanding is that there's some echo commands as part of ssh-agent's output that we don't want to be executed everytime the `.bashrc` is sourced?

Reply:
>* The commandLine variable could probably be a constant to make the rest of the code a little easier to read (like we did [here](https://github.com/devfile/devworkspace-operator/pull/1251/files#diff-f1f3498fd50bbc235d4d09f5f5485a91f5a4a5cfeb20b127c96fc849a30ba56bR30-R46))
>* It might be better to use $HOME instead of hard-coding /home/user/ since the users workspace might not be using the UDI. I'm not sure how common the ssh-agent binary is, but hopefully your implementation here could also work for other workspace images.
>* This is very subjective: I find using source is more readable than .. This is relevant IMO because we are modifying the users .bashrc.

done
> The only non-straightforward thing here is the ssh-agent | sed 's/^echo/#echo/' > $SSH_ENV_PATH. My understanding is that there's some echo commands as part of ssh-agent's output that we don't want to be executed everytime the .bashrc is sourced?

Right, we comment the `echo Agent pid ****;` line of the `ssh-agent` command output, so we have a clear terminal start. As an alternative we can add `>/dev/null` to the `source ${SSH_ENV_PATH}` bachrc command. WDYT?

Reply:
I actually like your current solution of not echoing the ` Agent pid ****`  over doing a `>/dev/null` since this would probably let any errors related to sourcing the ssh-agent environment variables get logged in the user's terminal and let them know something is not working correctly.

Reply:
Also @vinokurig it seems we're still referencing `/home/user/` in two liens of the postStart event: [here](https://github.com/devfile/devworkspace-operator/pull/1307/files#diff-12b01779c7cf67e18ef8d7bf46eb788bb6b640db0fa051804f2ef7b31a50d4d5R22) and [here](https://github.com/devfile/devworkspace-operator/pull/1307/files#diff-12b01779c7cf67e18ef8d7bf46eb788bb6b640db0fa051804f2ef7b31a50d4d5R27). Could you please change these to `$HOME`?